candle-nn/src/ops.rs

```diff
@@ -201,6 +201,46 @@ impl candle::CustomOp1 for SoftmaxLastDim {
         };
         Ok((dst, layout.shape().clone()))
     }
+
+    #[cfg(feature = "metal")]
+    fn metal_fwd(
+        &self,
+        storage: &candle::MetalStorage,
+        layout: &Layout,
+    ) -> Result<(candle::MetalStorage, Shape)> {
+        use candle::{backend::BackendStorage, DType};
+        let device = storage.device();
+        let command_buffer = device.command_buffer();
+        let kernels = device.kernels();
+        let name = match storage.dtype() {
+            DType::F32 => "softmax_float",
```

Comment:
The convention on the cuda side is to use names that are based on the rust type, e.g. `softmax_f32` etc. Using this we generate the name of the kernels automatically for most ops. Might be worth to be uniform here and rename these appropriately?

Reply:
Done.